packages/discovery-react-components/src/utils/document/__tests__/idUtils.spec.tsx

```diff
@@ -0,0 +1,49 @@
+import { getId, findElement, findElementIndex } from '../idUtils';
+
+const elements = [
+  { text: 'Koya Creative LLC', location: { begin: 469800, end: 469817 } },
+  { location: { begin: 282689, end: 282844 }, text: 'This Agreement may only be amended' }
+];
+
+const relation: any = [
+  {
+    type: 'invoice_parts',
+    allAttributeIds: ['1618_1621', '1876_1879', '2126_2195', '2435_2440']
+  },
+  {
+    type: 'invoice_parts',
+    allAttributeIds: ['1618_1621', '2949_2951', '3197_3268', '3508_3513']
+  }
+];
+
+describe('idUtils - function getId', () => {
+  it('gives back id from a valid item', () => {
+    expect(getId(elements[0])).toBe('469800_469817');
+  });
+
+  it('gives back undefined from an invalid item', () => {
+    const item: any = { text: 'Foo bar' };
+    expect(getId(item)).toBeUndefined();
+  });
```

Comment:
Hmm, this points out a possible issue with current implementation. _If_ there is invalid data, there could be multiple with all the same ID of `undefined`. Could this cause any problems? Would it be better to throw an error or somehow fail gracefully?

Reply:
I have not seen such a scenario/case till now, but yes it would give the same `undefined` id, (could cause issues on the UI, as it will always go to the 1st element, etc. )